codecommit-put-file-action/entrypoint.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -e
+
+# set aws-credentials
+mkdir -p ~/.aws
+touch ~/.aws/credentials
+
+echo "[default]
+aws_access_key_id = ${AWS_ACCESS_KEY_ID}
+aws_secret_access_key = ${AWS_SECRET_ACCESS_KEY}" > ~/.aws/credentials
+
```

Comment:
There should be some checks that these environment variables have been set as expected and required. You want to be sure that the values are correct to prevent making a bad commit.

Reply:
I realized that the `AWS S3 cp` GitHub Action doesn't check for environment variables being properly set either, but I think when `FILES` or `AWS_S3_BUCKET` is empty, the command will either do nothing or fail. I'm not sure if that's the case here; for example, it may make a commit with an empty commit message.

Reply:
yep that's a great point.. 

- seems like a check is required to see if the branch exists (else codecommit will create a new branch)
- empty commit message is a bad practice, but should it be illegal?

Reply:
> * empty commit message is a bad practice, but should it be illegal?

I agree that it's bad practice.

I didn't know `git` already disallows empty commit messages (unless `--allow-empty-message` is used). I'm not sure if `aws codecommit put-file` will workaround that. I think let's not do anything about it ourselves here; let `git` and/or `aws codecommit put-file` deal with it.

Reply:
codecommit actually allows empty commits but sticking to their original behavior makes sense for us